### PR TITLE
Update run-node.py

### PR DESCRIPTION
**Pull request by cordt-sei**
### Purpose of the Changes
addresses several issues and improvements:

- **Dynamic Version Control**: Implementing version control with automated fetching based on network, fetched from 'abci_info' RPC endpoint. Additional variables at top of the script to override versioning.

- **Database Backend Selection**: In response to recent updates which included additional database backend options, the script now prompts the user to choose their preferred database backend. This ensures compatibility and customization according to user needs.

- **Dependency Management**: Checks for and installs missing Python packages automatically or alerts to install [requests].

- **Enhanced Error Handling and Console Logs**: Added error handling just based on issues I ran into during this rework, added a few console messages during setup confirming selections.

- **"Advanced User Configs"**: Several variables are now near the top of the script for quick access to commonly modified parameters:

```
# advanced user configs
moniker = "pynode" # optional custom moniker for the node
trust_height_delta = 40000  # negative height offset for state sync
enable_unsafe_reset = True  # wipe database and keys before setup
version_override = False  # override version fetching. [if true, specify version(s) below]

# chain binary version ["version_override" must be true to use]
MAINNET_VERSION = "v3.9.0"
DEVNET_VERSION = "v5.2.2"
TESTNET_VERSION = "v5.2.2"
```

### Impact of Changes
Makes the script functional again and is easy to maintain throughout version changes.

### Testing
Tested on my local machine, pacific-1 + arctic-1 both using sei-db. Notifies if missing 'requests' package, prints network and version number upon selecting, node syncs without app hash.

### CODE REVIEW

File: scripts/run-node.py [hunk 1]
```diff
@@ -1,119 +1,176 @@
...
 
-# Mapping of env to chain_id
-ENV_TO_CHAIN_ID = {
-    "local": None,
-    "devnet": "arctic-1",
-    "testnet": "atlantic-2"
+# req packages
+dependencies = ['requests']
+
+for package in dependencies:
```
Comment (philipsu522) [inline scripts/run-node.py:14]: nice

File: scripts/run-node.py [hunk 2]
```diff
@@ -1,119 +1,176 @@
...
+MAINNET_VERSION = "v3.9.0"
+DEVNET_VERSION = "v5.2.2"
+TESTNET_VERSION = "v5.2.2"
+
+# map env to chain ID and optional manual version override
+ENV_TO_CONFIG = {
+    "local": {"chain_id": None, "version": "latest"},
+    "devnet": {"chain_id": "arctic-1", "version": DEVNET_VERSION},
+    "testnet": {"chain_id": "atlantic-2", "version": TESTNET_VERSION},
+    "mainnet": {"chain_id": "pacific-1", "version": MAINNET_VERSION}
```
Comment (philipsu522) [inline scripts/run-node.py:37]: now that we're fetching from abci_info we could remove this (and probably should? to avoid app-hashes)

**Review by philipsu522 — APPROVED** [review]
this is great!

File: scripts/run-node.py (hunk 2 above)
Comment (Cordtus) [reply scripts/run-node.py:37]: I left that in there for manual override incase someone wants a specific version for whatever reason, testing or debug etc.

It's actually ignored unless the option above is set to true

**Pull request merged**